docs/relational_table/relational-table-schema.md

```diff
@@ -0,0 +1,35 @@
+# Relational Table Schema
+
+We introduce the rough cell-based encoding format in [relational states](storing-state-using-relational-table.md#cell-based-encoding)
+
+In this doc, we will take HashAgg with extreme state (`max`, `min`) or value state (`sum`, `count`) for example, and introduce a more detailed design for the internal table schema.
+
+[Code](https://github.com/singularity-data/risingwave/blob/7f9ad2240712aa0cfe3edffb4535d43b42f32cc5/src/frontend/src/optimizer/plan_node/logical_agg.rs#L144)
+
+## Table id
+For all relational table states, the keyspace must start with `table_id`. This is a globally unique id allocated in meta. Meta is responsible for traversing the Plan Tree and calculating the total number of Relational Tables needed. For example, the Hash Join Operator needs 2, one for the left table and one for the right table. The number of tables needed for Agg depends on the number of agg calls.
+
+## Value State (Sum, Count)
+Query example:
+```sql
+select sum(v2), count(v3) from t group by v1 
+```
+
+This query will need to initiate 2 Relational Tables. The schema is `table_id/group_key/column_id`.
+
+## Extreme State (Max, Min)
```

Comment:
The corresponding concept is actually named `TableState`. But since we have only one implementation of `TableState` now, that is , `ExtremeState`, this also LGTM.

https://github.com/singularity-data/risingwave/blob/1bb848be3db39ade1a759d4f945498d607c985cb/src/stream/src/executor/managed_state/aggregation/extreme.rs#L101